### PR TITLE
Add an overview for the exported metrics

### PR DESCRIPTION
**Pull request by fieteboerner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,3 +119,27 @@ scrape_configs:
     static_configs:
       - targets: ['localhost:9205']
 ```
+
+### Exported metrics
+
+This are the metrics exported by the `nextcloud-exporter`
```
Comment (xperimental) [inline README.md:125]:
```suggestion
These metrics are exported by `nextcloud-exporter`:
```

File: README.md [hunk 2]
```diff
@@ -119,3 +119,27 @@ scrape_configs:
...
+
+| name                                   | description                                                            |
+|----------------------------------------|------------------------------------------------------------------------|
+| nextcloud_active_users_total           | Number of active users for the last five minutes                       |
+| nextcloud_apps_installed_total         | Number of currently installed apps                                     |
+| nextcloud_apps_updates_available_total | Number of apps that have available updates                             |
+| nextcloud_auth_errors_total            | Counts number of authentication errors encountered by the collector    |
+| nextcloud_database_size_bytes          | Size of database in bytes as reported from engine                      |
+| nextcloud_exporter_info                | Contains meta informatoin of the exporter. Value is always 1           |
```
Comment (xperimental) [inline README.md:134]:
```suggestion
| nextcloud_exporter_info                | Contains meta information of the exporter. Value is always 1.           |
```

File: README.md [hunk 3]
```diff
@@ -119,3 +119,27 @@ scrape_configs:
...
+| nextcloud_files_total                  | Number of files served by the instance                                 |
+| nextcloud_free_space_bytes             | Number of available disk space on the instance in bytes                |
+| nextcloud_php_info                     | Contains meta information about PHP as labels. Value is always 1       |
```
Comment (xperimental) [inline README.md:137]:
```suggestion
| nextcloud_php_info                     | Contains meta information about PHP as labels. Value is always 1.       |
```

File: README.md [hunk 4]
```diff
@@ -119,3 +119,27 @@ scrape_configs:
...
+| nextcloud_php_memory_limit_bytes       | Configured PHP memory limit in bytes                                   |
+| nextcloud_php_upload_max_size_bytes    | Configured maximum upload size in bytes                                |
+| nextcloud_scrape_errors_total          | Number of scrape errors from this collector                            |
+| nextcloud_shares_federated_total       | Number of federated shares by directions `sent` / `received`           |
+| nextcloud_shares_total                 | Number of shares by type: <br> `authlink`: shared password protected links <br> `group`: shared groups <br>`link`: all shared links <br> `user`: shared users |
+| nextcloud_system_info                  | Contains meta information about Nextcloud as labels. Value is always 1 |
```
Comment (xperimental) [inline README.md:143]:
```suggestion
| nextcloud_system_info                  | Contains meta information about Nextcloud as labels. Value is always 1. |
```

File: README.md [hunk 5]
```diff
@@ -119,3 +119,27 @@ scrape_configs:
...
+| nextcloud_up                           | Shows if nextcloud instance is up `0` down / `1` up                    |
```
Comment (xperimental) [inline README.md:144]: I have seen this metric misunderstood a few times already and thought about rewording the help text to help this, maybe you can help me here: The metric does not actually show "if the Nextcloud server is up" but instead shows if the exporter is able to extract data from it, which can have multiple causes (for example: the server is down, the network is broken, the credentials are wrong). It is called `nextcloud_up` to be similar to the `up` metric created by Prometheus for all targets.

File: README.md [hunk 6]
```diff
@@ -119,3 +119,27 @@ scrape_configs:
...
+| name                                   | description                                                            |
+|----------------------------------------|------------------------------------------------------------------------|
+| nextcloud_active_users_total           | Number of active users for the last five minutes                       |
+| nextcloud_apps_installed_total         | Number of currently installed apps                                     |
+| nextcloud_apps_updates_available_total | Number of apps that have available updates                             |
+| nextcloud_auth_errors_total            | Counts number of authentication errors encountered by the collector    |
+| nextcloud_database_size_bytes          | Size of database in bytes as reported from engine                      |
+| nextcloud_exporter_info                | Contains meta informatoin of the exporter. Value is always 1           |
+| nextcloud_files_total                  | Number of files served by the instance                                 |
+| nextcloud_free_space_bytes             | Number of available disk space on the instance in bytes                |
```
Comment (xperimental) [inline README.md:136]:
```suggestion
| nextcloud_free_space_bytes             | Free disk space in data directory in bytes                |
```

**Review by xperimental — CHANGES_REQUESTED** [review]
Thanks for the contribution, this seems like an idea that could help a few people. :tada: 

In addition to the comments on the source I am also thinking about whether the generic metrics (`go_`, `process_`, `promhttp_`) should be mentioned. I wouldn't list them in a table, but only mention that the exporter not only exports the metrics listed in the table but also generic metrics about the runtime and process.

File: README.md (hunk 5 above)
Comment (fieteboerner) [reply README.md]: Good to know. I misunderstood this metric too. 

I ve updated the description according to your explanation.

**Comment by fieteboerner** [comment]
Happy to help.

According to your comment:
> In addition to the comments on the source I am also thinking about whether the generic metrics (go_, process_, promhttp_) should be mentioned. I wouldn't list them in a table, but only mention that the exporter not only exports the metrics listed in the table but also generic metrics about the runtime and process.

I dont know about these metrics. I couldnt find them in prometheus. - Probable it is better if you could mention them, before I write something I dont know/understand.

**Comment by fieteboerner** [comment]
reopen pr, because I closed it accidentially

File: README.md (hunk 5 above)
Comment (xperimental) [reply README.md]: Let's go with that explanation then :+1:

**Review by xperimental — APPROVED** [review]
I've updated the help text exported with the metric for those that were changed. I think it's fine now, can do improvements to the table later.

**Pull request merged**